projects/ngx-formentry/src/form-entry/helpers/js-expression-helper.ts

```diff
@@ -164,6 +166,68 @@ export class JsExpressionHelper {
     return height && weight && refSectionObject ? formattedSDValue : null;
   }
 
+  getAgePosition(age) {
+    if (age >= 70) {
+      return 0;
+    } else if (age >= 65) {
+      return 1;
+    } else if (age >= 60) {
+      return 2;
+    } else if (age >= 55) {
+      return 3;
+    } else if (age >= 50) {
+      return 4;
+    } else if (age >= 45) {
+      return 5;
+    }
+    return 6;
+  }
+
+  getSBPPosition(sbp) {
+    if (sbp >= 180) {
+      return 0;
+    } else if (sbp >= 160) {
+      return 1;
+    } else if (sbp >= 140) {
+      return 2;
+    } else if (sbp >= 120) {
+      return 3;
+    } else if (sbp >= 50) {
+      return 4;
+    } else if (sbp >= 45) {
+      return 5;
+    }
+    return 6;
+  }
+
+  getBMIPosition(bmi) {
+    if (bmi >= 35) {
+      return 4;
+    } else if (bmi >= 30) {
+      return 3;
+    } else if (bmi >= 25) {
+      return 2;
+    } else if (bmi >= 20) {
+      return 1;
+    }
+    return 0;
+  }
+
+  // Order Sex -> Age -> SBP -> Smoker -> BMI
+  calculateBMI(height, weight) {
+    return weight / (height * height);
+  }
+
+  calcNonLabCVDRisk({ age, sex, smoker, height, weight, sbp }) {
+    const bmi = this.calculateBMI(height, weight);
+    // Variables
+    const baseDataset = sex ? manDataset : womanDataset;
+
+    const riskScore = baseDataset[this.getAgePosition(age)][this.getSBPPosition(sbp)][smoker ? 1 : 0][this.getBMIPosition(bmi)];
```

Comment:
So, I chose the order that things were specified in to make it as easy as possible to transcribe the charts into arrays. Messing with this order means that you can't just read the arrays from the chart.

In particular, when reading the chart, we first look at sex (I suppose it doesn't matter if these are two separate multi-dimensional arrays or not, but it seems kinda silly to me to make them separate arrays when we have other factors). Then, we look at smoking status, then age, then SBP, then BMI. Switching the order of these means that the arrays that we use here will not easily match up to the numbers on the chart.

Reply:
Sure Ian, I just realised that I messed some order and I am going to fix that ASAP.